File: backend/app/schemas/transactions.py
from datetime import date, datetime
from decimal import Decimal
from pydantic import BaseModel, Field


class TransactionBase(BaseModel):
    account: str = Field(min_length=1, max_length=50)
    asset_type: str = Field(min_length=1, max_length=50)
    code: str = Field(min_length=1, max_length=50)
    name: str = Field(min_length=1, max_length=200)
    side: str = Field(pattern="^(buy|sell|dividend)$")
    trade_date: date
    quantity: Decimal = Field(gt=0)
    price: Decimal = Field(ge=0)
    fee: Decimal = Field(default=Decimal("0"), ge=0)
    currency: str = Field(default="CNY", max_length=10)
    notes: str | None = None


class TransactionCreate(TransactionBase):
    pass


class TransactionResponse(TransactionBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/transactions.ts
import api from './request'

export type TransactionSide = 'buy' | 'sell' | 'dividend' | 'deposit' | 'withdraw'

/** 资金流水方向列表 */
export const CASH_FLOW_SIDES: TransactionSide[] = ['deposit', 'withdraw']
/** 持仓交易方向列表 */
export const ASSET_SIDES: TransactionSide[] = ['buy', 'sell', 'dividend']

export interface Transaction {
  id: number
  account: string
  asset_type: string
  code: string
  name: string
  side: TransactionSide
  trade_date: string
  quantity: number
  price: number
  fee: number
  currency: string
  notes: string | null
  created_at: string
  updated_at: string
}

export type TransactionCreate = Omit<Transaction, 'id' | 'created_at' | 'updated_at'>

export const getTransactions = () => api.get<Transaction[]>('/transactions')
export const createTransaction = (data: TransactionCreate) => api.post<Transaction>('/transactions', data)
export const deleteTransaction = (id: number) => api.delete(`/transactions/${id}`)
